src/modules/emote_menu/components/Emotes.jsx

```diff
@@ -0,0 +1,89 @@
+import React, {useCallback, useEffect, useRef} from 'react';
+import VirtualizedList from './VirtualizedList.jsx';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/emotes.module.css';
+import Emote from './Emote.jsx';
+import EmoteSearch from '../search.js';
+
+const ROW_HEIGHT = 36;
+const WINDOW_HEIGHT = 308;
+const TOTAL_COLS = 7;
+
+function Emotes({onClick, onSelect, focus, onFocus}) {
+  const wrapperRef = useRef(null);
+
+  const renderRow = useCallback(
+    ({key, style, index}) => {
+      const row = emoteStore.getRow(index);
+      return emoteStore.isHeader(index) ? (
+        <div key={key} style={style} className={styles.header}>
+          {row.displayName}
+        </div>
+      ) : (
+        <div key={key} style={style} className={styles.row}>
+          {row.map((emote) => (
+            <Emote emote={emote} onClick={() => onClick(emote)} onMouseOver={() => onSelect(emote)} />
+          ))}
+        </div>
+      );
+    },
+    [onSelect, onClick]
+  );
+
+  const headerChange = useCallback((index) => {
```

Comment:
nit: try to keep naming consistent. you seem to use `onHeaderChange`, but `headerChange` here and `onSelect` above. let's normalize on `onName` for props and `handleName` for functions we're passing in.

```
onHeaderChange={handleHeaderChange}
```